File: codeforces/714B/main.hs
-- ID:  714B (Filya and Homework)
-- URL: http://codeforces.com/problemset/problem/714/B

import Data.List

getInt = read `fmap` getLine :: IO Int

main = do
	n <- getInt
	xs <- (map read . take n . words) `fmap` getContents :: IO [Int]
	putStrLn $ (judge . collect) xs

collect xs = f xs [] where
	f [] acc = Just acc
	f (x:xs) acc
		| x `elem` acc    = f xs acc
		| length acc == 3 = Nothing
		| otherwise       = f xs (x:acc)

judge Nothing = "NO"
judge (Just [x]) = "YES"
judge (Just [x,y]) = "YES"
judge (Just [x,y,z])
	| arithmeticSeq (sort [x,y,z]) = "YES"
	| otherwise = "NO"

arithmeticSeq [x,y,z]
	| y-x == z-y = True
	| otherwise  = False

File: codeforces/714B/main.cc
// ID:  714B (Filya and Homework)
// URL: a

//#include <stdio.h>
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

int main() {
	int n, x;
	vector<int> xs;

	cin >> n;
	for(int i=0; i<n; i++){
		cin >> x;
		if(find(xs.begin(), xs.end(), x) == xs.end()){
			xs.push_back(x);
		}
		if(xs.size() >= 4){
			cout << "NO" << endl;
			return 0;
		}
	}
	sort(xs.begin(), xs.end());

	if(xs.size() == 1){
		cout << "YES" << endl;
		return 0;
	}

	for(int i=1; i<xs.size(); i++){
		if(xs[i] - xs[i-1] != xs[1] - xs[0]){
			cout << "NO" << endl;
			return 0;
		}
	}

	cout << "YES" << endl;

	return 0;
}
